docs/guides/plugin_dev/custom_model.rst

```diff
@@ -0,0 +1,175 @@
+.. _custom_model_builder:
+
+Custom Model Builders
+=====================
+
+
+The main goal of your plugin is to be able to build and update model instances for your
+own software. As a reminder, the figure below illustrates the process of model building
+in HydroMT:
+
+.. figure:: ../../_static/model_building_process.png
+   :align: center
+
+   Model building process in HydroMT.
+
+In this section, we will detail the bottom part of the schematic above ie how to initialise your HydroMT Model class for your plugin, the main properties
+and setup methods and how to work with its model components. The top part of the schematic (data preparation) is taken care of by the core of HydroMT.
+
+Initialisation
+^^^^^^^^^^^^^^
+
+Most of the functionality necessary to initialise models is already taken care of in
+Core, therefore if you inherit from the core `Model` class you should not have to
+implement much or even any of the basic functionalities.
+
+If you wish to provide default behaviour for your users, such as providing default
+components that your model will always need, you can do so by overriding the `__init__`
+function on your model like so:
+
+```python
+class AwesomeModel:
+   def __init__(self, ...):
+      super().__init__(...)
+      self.add_component("extra_grid", GridComponent)
+      # extra customisation code here...
+```
+
+Please make sure to call the `super().__init__` function before you do anything else to
+setup your class so that the base functionality also get's run properly.
+
+Additional Model properties
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+You can have a look at the :ref:`Model API documentation <model_api>` to find out all other properties of the ``Model`` class and
+the other subclasses. Apart from the components here are a couple of useful properties to use when developing your plugin:
+
+- :py:attr:`~Model.root`: path to the root folder of the model instance you are working with.
+- :py:attr:`~Model.crs`: the reference coordinate system (pyproj.CRS) of the model.
+- :py:attr:`~Model.region`: A shortcut to retrieve region data from the designated
+     spacial component
+- :py:attr:`~Model.data_catalog`: the current data catalog you can use to add data to your model.
+- :py:attr:`~Model.logger`: the HydroMT logger object for your model instance.
+
+
+.. _plugin_setup:
```

Comment:
I would make this tag more specific, now that there is more plugin behaviour in the rest of the code